server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -682,5 +641,53 @@ long getInitialTermBeforeJoiningALeader() {
         public void close() {
             assertTrue(openPersistedStates.remove(this));
         }
+
+        @Override
+        public void getLatestStoredState(ActionListener<ClusterState> listener) {
+            ActionListener.completeWith(listener, () -> {
+                var latestClusterState = sharedStore.getLatestClusterState();
+                if (latestClusterState == null) {
+                    return null;
+                }
+
+                if (isLatestAcceptedStateStale(latestClusterState) == false) {
+                    return null;
+                }
+
+                if (latestClusterState.term() > currentTerm) {
+                    return null;
+                }
+
+                return ClusterStateUpdaters.recoverClusterBlocks(
+                    ClusterStateUpdaters.addStateNotRecoveredBlock(
+                        ClusterState.builder(new ClusterName("elasticsearch"))
+                            .metadata(
+                                Metadata.builder(latestClusterState.state())
+                                    .coordinationMetadata(
+                                        new CoordinationMetadata(
+                                            latestClusterState.term(),
+                                            // Keep the previous configuration so the assertions don't complain about
+                                            // a different configuration, we'll change it right away
+                                            latestAcceptedState.getLastAcceptedConfiguration(),
+                                            latestAcceptedState.getLastCommittedConfiguration(),
```

Comment:
We could override this here too I think? (capturing `localNode` from the constructor)

```suggestion
                                            CoordinationMetadata.VotingConfiguration.of(localNode),
                                            CoordinationMetadata.VotingConfiguration.of(localNode),
```

Reply:
We still need to change the voting configuration when the new master has the latest state.

Reply:
True. Maybe we should just fix that in the election-winning publication always (possibly via some different mechanism since it's not really related to reading it from the store). No need to do that here, we can follow up at a later date.

Reply:
We already do that in https://github.com/elastic/elasticsearch/blob/5dc2541675b897f1d64ceec514f76c22a5c3bf63/server/src/main/java/org/elasticsearch/cluster/coordination/NodeJoinExecutor.java#L270 maybe you're referring about something else?

Reply:
D'oh yes that's what I meant.